tests/unit/test_auth_middleware.py

```
@@ -0,0 +1,330 @@
+
+
+class TestAuthenticatedEndpoints:
+    """Tests for endpoints requiring authentication."""
+
+    def test_missing_token_returns_401(self, client: TestClient):
+        """Request without token should return 401."""
+        response = client.get("/api/v1/user/me")
+        assert response.status_code == 401
+        assert "Missing authentication token" in response.json()["detail"]
+
+    def test_valid_token_returns_user(self, client: TestClient, mock_settings):
```

Ruff (0.14.8)

Unused method argument: `mock_settings`

(ARG002)